### PR TITLE
[nanonng] eliminate code warnings

### PR DESCRIPTION
**Pull request by RanMaoyi**
Only eliminate on x86_64 and Standard compilation options

### CODE REVIEW

**Review by OdyWayne — APPROVED** [review]

File: include/nng/protocol/mqtt/mqtt_parser.h
Comment (xinyi-xs) [inline include/nng/protocol/mqtt/mqtt_parser.h:1]: If we need to consider cross-platform compatibility.

File: include/nng/protocol/mqtt/mqtt_parser.h [hunk 1]
```diff
@@ -13,11 +13,11 @@
 // Do not change to %lu! just suppress the warning of the compiler!
 #define DISCONNECT_MSG          \
 	"{\"username\":\"%s\"," \
-	"\"ts\":%llu,\"reason_code\":\"%x\",\"client_id\":\"%s\",\"IPv4\":\"%s\"}"
+	"\"ts\":%lu,\"reason_code\":\"%x\",\"client_id\":\"%s\",\"IPv4\":\"%s\"}"
 
 #define CONNECT_MSG                                                           \
 	"{\"username\":\"%s\", "                                              \
-	"\"ts\":%llu,\"proto_name\":\"%s\",\"keepalive\":%d,\"return_code\":" \
+	"\"ts\":%lu,\"proto_name\":\"%s\",\"keepalive\":%d,\"return_code\":" \
```
Comment (JaylinYu) [inline include/nng/protocol/mqtt/mqtt_parser.h:20]: this gonna cause crash on aarch64
read the comments I left

File: src/mqtt/protocol/exchange/exchange_server.c [hunk 2]
```diff
@@ -330,7 +330,7 @@ exchange_sock_recv(void *arg, nni_aio *aio)
 			return;
 		}
 	} else {
-		if (tss[2] == NULL || tss[2] == 0) {
+		if (tss[2] == 0) {
```
Comment (JaylinYu) [inline src/mqtt/protocol/exchange/exchange_server.c:333]: ?different logic?

File: src/mqtt/transport/tcp/mqtt_tcp.c [hunk 3]
```diff
@@ -900,8 +900,15 @@ mqtt_tcptran_pipe_send_start(mqtt_tcptran_pipe *p)
 			if (qos > 0)
 				p->sndmax --;
 			if (qos > p->qosmax) {
-				p->qosmax == 1? (*header &= 0XF9) & (*header |= 0X02): NNI_ARG_UNUSED(*header);
-				p->qosmax == 0? *header &= 0XF9: NNI_ARG_UNUSED(*header);
+				if (p->qosmax == 0 || p->qosmax == 1) {
```
Comment (JaylinYu) [inline src/mqtt/transport/tcp/mqtt_tcp.c:903]: I pretty much love the original way
And it performs better

File: src/supplemental/mqtt/mqtt_codec.c [hunk 4]
```diff
@@ -4192,13 +4192,6 @@ encode_properties(nni_msg *msg, property *prop, uint8_t cmd)
 	return 0;
 }
 
-static nni_proto_msg_ops proto_ops = {
```
Comment (JaylinYu) [inline src/supplemental/mqtt/mqtt_codec.c:4195]: This is an un-implemented feature
just comment it out

File: src/supplemental/nanolib/parquet/parquet.cc [hunk 5]
```diff
@@ -821,7 +821,6 @@ parquet_read(conf_parquet *conf, char *filename, vector<uint64_t> keys)
 			int64_t values_read = 0;
 			int64_t rows_read   = 0;
 			int16_t definition_level;
-			int16_t repetition_level;
```
Comment (JaylinYu) [inline src/supplemental/nanolib/parquet/parquet.cc:824]: is it useful ? @xinyi-xs

File: src/supplemental/util/idhash.c [hunk 6]
```diff
@@ -64,12 +64,12 @@ nng_id_alloc(nng_id_map *map, uint64_t *id, void *val)
 void
 nng_id_map_foreach(nng_id_map *map, void (*cb)(void *id, void *value))
 {
-	nni_id_map_foreach(map, cb);
+	nni_id_map_foreach((nni_id_map *)map, cb);
```
Comment (JaylinYu) [inline src/supplemental/util/idhash.c:67]: check upstream

**Review by JaylinYu — CHANGES_REQUESTED** [review]
most parts are fine.
Plz Resolve issue from commentator

File: src/supplemental/nanolib/parquet/parquet.cc (hunk 5 above)
Comment (xinyi-xs) [reply src/supplemental/nanolib/parquet/parquet.cc:824]: It may not be useful now, but it could come in handy in the future.

**Review by JaylinYu — APPROVED** [review]
now world is cleaned

**Review by wanghaEMQ — COMMENTED** [review]
Cool!

**Pull request merged**